shell/platform/windows/client_wrapper/include/flutter/flutter_engine.h

```diff
@@ -62,6 +62,9 @@ class FlutterEngine : public PluginRegistry {
   FlutterDesktopPluginRegistrarRef GetRegistrarForPlugin(
       const std::string& plugin_name) override;
 
+  // Returns the texture registrar for this engine.
+  FlutterDesktopTextureRegistrarRef GetTextureRegistrar();
```

Comment:
We should expose a wrapper, not the raw type. The only reason the plugin registrar is the raw type is that the use case for it is to pass to a plugin registration function, which is a C API (as it's a dll interface). This should be like the messenger or the view, where the wrapper vends a wrapper.

Reply:
I don't currently have time to write a wrapper for `FlutterDesktopTextureRegistrar`, nor do I think that should that be in scope this PR.  I can remove this from `FlutterEngine` and just call `FlutterDesktopEngineGetTextureRegistrar` somewhere, but I'm really not sure what the proper place for that would be. Any suggestion is appreciated.

Reply:
A direct test would a test written against the C API directly. It doesn't look like there are any at the moment, but I don't remember if there's a reason doing that is problematic, or it was just never backfilled. It may just need some build plumbing to make it possible to launch the engine from a test (IIRC that was done for macOS, but I'm not sure if Linux and Windows have it.)

@cbracken I know you have a plan around doing some more testing of the desktop embeddings; is that part of that plan?

Reply:
The C api is currently tested indirectly through the client wrapper. Since there's no wrapper for texture registraryet I added a test that just calls the C API. (`TEST(FlutterEngineTest, GetTextureRegistrar)`)

Reply:
> @cbracken I know you have a plan around doing some more testing of the desktop embeddings; is that part of that plan?

That is indeed part of the plan. The person working on it has put that work on hold for the moment. You're correct that some progress was made on the macOS side of things.  Filed flutter/flutter#87299 for this.

Reply:
@cbracken, is that something that should block this PR? It just fixes the method signature (otherwise it can't be called as the underlying implementation is not exported)